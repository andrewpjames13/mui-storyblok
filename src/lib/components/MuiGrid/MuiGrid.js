import React from 'react';
import Grid from '../Grid/Grid';
import MuiAppBar from '../MuiAppBar/MuiAppBar';
import MuiMobileStepper from '../MuiMobileStepper/MuiMobileStepper';
import MuiHeroHeader from '../MuiHeroHeader/MuiHeroHeader';
import MuiCircularProgress from '../MuiCircularProgress/MuiCircularProgress';
import MuiGridList from '../MuiGridList/MuiGridList';
import MuiButtonRedirect from '../MuiButtonRedirect/MuiButtonRedirect';
import MuiCard from '../MuiCard/MuiCard';
import MuiTypography from '../MuiTypography/MuiTypography';
import MuiMenu from '../MuiMenu/MuiMenu';
import MuiIcon from '../MuiIcon/MuiIcon';
import MuiExpansionPanel from '../MuiExpansionPanel/MuiExpansionPanel';
import MuiList from '../MuiList/MuiList';
import MuiTable from '../MuiTable/MuiTable';
import MuiPaginationTable from '../MuiPaginationTable/MuiPaginationTable';
import MuiContactButton from '../MuiContactButton/MuiContactButton';
import MuiImage from '../MuiImage/MuiImage';
import BlokForm from '../BlokForm/BlokForm';

const MuiGrid = (props) => {
  const gridItemComponents = {
    MuiHeroHeader,
    MuiAppBar,
    MuiMobileStepper,
    MuiCircularProgress,
    MuiGridList,
    MuiButtonRedirect,
    MuiCard,
    MuiTypography,
    MuiMenu,
    MuiIcon,
    MuiExpansionPanel,
    MuiList,
    MuiTable,
    MuiPaginationTable,
    MuiContactButton,
    MuiImage,
    BlokForm,
  };

  return (
    <Grid
      {...props}
      gridItemComponents={gridItemComponents}
    />
  );
};

export default MuiGrid;
