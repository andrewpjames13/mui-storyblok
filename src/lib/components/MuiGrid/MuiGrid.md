MuiGrid Spacing Prop:

```js noeditor
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import FormLabel from '@material-ui/core/FormLabel';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import RadioGroup from '@material-ui/core/RadioGroup';
import Radio from '@material-ui/core/Radio';
import Paper from '@material-ui/core/Paper';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  paper: {
    height: 140,
    width: 100,
  },
  control: {
    padding: theme.spacing(2),
  },
}));

function SpacingGrid() {
  const [spacing, setSpacing] = React.useState(2);
  const classes = useStyles();

  const handleChange = (event) => {
    setSpacing(Number(event.target.value));
  };

  return (
    <Grid container className={classes.root} spacing={2}>
      <Grid item xs={12}>
        <Grid container justify="center" spacing={spacing}>
          {[0, 1, 2].map((value) => (
            <Grid key={value} item>
              <Paper className={classes.paper} />
            </Grid>
          ))}
        </Grid>
      </Grid>
      <Grid item xs={12}>
        <Paper className={classes.control}>
          <Grid container>
            <Grid item>
              <FormLabel>spacing</FormLabel>
              <RadioGroup
                name="spacing"
                aria-label="spacing"
                value={spacing.toString()}
                onChange={handleChange}
                row
              >
                {[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((value) => (
                  <FormControlLabel
                    key={value}
                    value={value.toString()}
                    control={<Radio />}
                    label={value.toString()}
                  />
                ))}
              </RadioGroup>
            </Grid>
          </Grid>
        </Paper>
      </Grid>
    </Grid>
  );
}

<SpacingGrid />
```

MuiGrid Interactive Example:
Props -> direction, justify, and alignItems:

```js noeditor
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import FormLabel from '@material-ui/core/FormLabel';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import RadioGroup from '@material-ui/core/RadioGroup';
import Radio from '@material-ui/core/Radio';
import Paper from '@material-ui/core/Paper';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  paper: {
    height: 140,
    width: 100,
    backgroundColor: '#AAAAAA',
  },
  control: {
    padding: theme.spacing(2),
  },
}));

function InteractiveGrid() {
  const [direction, setDirection] = React.useState('row');
  const [justify, setJustify] = React.useState('center');
  const [alignItems, setAlignItems ] = React.useState('center');
  const classes = useStyles();

  const handleDirection = (event) => {
    const { value } = event.target;
    setDirection(value);
  };

  const handleJustify = (event) => {
    const { value } = event.target;
    setJustify(value);
  };

  const handleAlign = (event) => {
    const { value } = event.target;
    setAlign(value);
  };

  return (
    <Grid container className={classes.root} spacing={2}>
      <Grid item xs={12}>
        <Grid container justify={justify} direction={direction} alignItems={alignItems}>
          {[0, 1, 2].map((value) => (
            <Grid key={value} item>
              <span>
                <h1>{value}</h1>
                <Paper className={classes.paper} elevation={10} />
              </span>
            </Grid>
          ))}
        </Grid>
      </Grid>
      <Grid item xs={12}>
        <Paper className={classes.control}>
          <Grid container>
            <Grid item>
              <FormLabel>direction</FormLabel>
              <RadioGroup
                name="direction"
                aria-label="direction"
                onChange={handleDirection}
                row
              >
                {['row', 'row-reverse', 'column', 'column-reverse'].map((value) => (
                  <FormControlLabel
                    key={value}
                    value={value}
                    control={<Radio />}
                    label={value}
                  />
                ))}
              </RadioGroup>
            </Grid>
            <Grid item>
              <FormLabel>justify</FormLabel>
              <RadioGroup
                name="justify"
                aria-label="justify"
                onChange={handleJustify}
                row
              >
                {['flex-start', 'center', 'flex-end', 'space-between', 'space-around', 'space-evenly'].map((value) => (
                  <FormControlLabel
                    key={value}
                    value={value}
                    control={<Radio />}
                    label={value}
                  />
                ))}
              </RadioGroup>
            </Grid>
            <Grid item>
              <FormLabel>alignItems</FormLabel>
              <RadioGroup
                name="alignItems"
                aria-label="alignItems"
                onChange={handleAlign}
                row
              >
                {['flex-start', 'center', 'flex-end', 'stretch', 'baseline'].map((value) => (
                  <FormControlLabel
                    key={value}
                    value={value}
                    control={<Radio />}
                    label={value}
                  />
                ))}
              </RadioGroup>
            </Grid>
          </Grid>
        </Paper>
      </Grid>
    </Grid>
  );
}

<InteractiveGrid />

```