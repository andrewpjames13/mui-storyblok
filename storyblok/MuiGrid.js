module.exports = {
  "name": "MuiGrid",
  "display_name": null,
  "updated_at": "2020-06-02T22:36:19.257Z",
  "schema": {
    "alignContent": {
      "type": "option",
      "pos": 0,
      "use_uuid": true,
      "options": [
        {
          "value": "stretch",
          "name": "stretch"
        },
        {
          "value": "center",
          "name": "center"
        },
        {
          "value": "flex-start",
          "name": "flex-start"
        },
        {
          "value": "flex-end",
          "name": "flex-end"
        },
        {
          "value": "space-between",
          "name": "space-between"
        },
        {
          "value": "space-around",
          "name": "space-around"
        }
      ],
      "required": true,
      "default_value": "stretch",
      "description": "   * mui prop: 'stretch'| 'center'| 'flex-start'| 'flex-end'| 'space-between'| 'space-around'    * Defines the align-content style property. It's applied for all screen sizes."
    },
    "alignItems": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "flex-start",
          "name": "flex-start"
        },
        {
          "value": "center",
          "name": "center"
        },
        {
          "value": "flex-end",
          "name": "flex-end"
        },
        {
          "value": "stretch",
          "name": "stretch"
        },
        {
          "value": "baseline",
          "name": "baseline"
        }
      ],
      "default_value": "center",
      "required": true,
      "pos": 1,
      "description": "   * mui prop: 'flex-start' | 'center' | 'flex-end' | 'stretch' | 'baseline'    * Defines the align-items style property. It's applied for all screen sizes."
    },
    "rootClass": {
      "type": "options",
      "options": [
        {
          "value": "background-color: black",
          "name": "backgroundColor"
        },
        {
          "value": "color: white",
          "name": "color"
        },
        {
          "value": "margin-top: 50px",
          "name": "margin-top"
        },
        {
          "value": "padding-top: 100px; padding-bottom: 100px",
          "name": "paddingStrip"
        }
      ],
      "pos": 2
    },
    "direction": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "row",
          "name": "row"
        },
        {
          "value": "row-reverse",
          "name": "row-reverse"
        },
        {
          "value": "column",
          "name": "column"
        },
        {
          "value": "column-reverse",
          "name": "column-reverse"
        }
      ],
      "default_value": "row",
      "required": true,
      "description": "   * mui prop: 'stretch'| 'center'| 'flex-start'| 'flex-end'| 'space-between'| 'space-around'    * Defines the align-content style property. It's applied for all screen sizes.",
      "pos": 3
    },
    "justify": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "flex-start",
          "name": "flex-start"
        },
        {
          "value": "center",
          "name": "center"
        },
        {
          "value": "flex-end",
          "name": "flex-end"
        },
        {
          "value": "space-between",
          "name": "space-between"
        },
        {
          "value": "space-around",
          "name": "space-around"
        },
        {
          "value": "space-evenly",
          "name": "space-evenly"
        }
      ],
      "default_value": "center",
      "required": true,
      "pos": 4,
      "description": "   * mui prop: 'flex-start' | 'center' | 'flex-end' | 'space-between' | 'space-around' | 'space-evenly'    * Defines the justify-content style property. It is applied for all screen sizes."
    },
    "spacing": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "0",
          "name": "0"
        },
        {
          "value": "1",
          "name": "1"
        },
        {
          "value": "2",
          "name": "2"
        },
        {
          "value": "3",
          "name": "3"
        },
        {
          "value": "4",
          "name": "4"
        },
        {
          "value": "5",
          "name": "5"
        },
        {
          "value": "6",
          "name": "6"
        },
        {
          "value": "7",
          "name": "7"
        },
        {
          "value": "8",
          "name": "8"
        },
        {
          "value": "9",
          "name": "9"
        },
        {
          "value": "10",
          "name": "10"
        }
      ],
      "default_value": "2",
      "required": true,
      "pos": 8
    },
    "wrap": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "nowrap",
          "name": "nowrap"
        },
        {
          "value": "wrap",
          "name": "wrap"
        },
        {
          "value": "wrap-reverse",
          "name": "wrap-reverse"
        }
      ],
      "required": true,
      "default_value": "wrap",
      "description": "Defines the flex-wrap style property. It's applied for all screen sizes.",
      "pos": 9
    },
    "content": {
      "type": "bloks",
      "maximum": "",
      "restrict_components": true,
      "component_whitelist": [
        "MuiGridItem",
        "MuiAppBar",
        "MuiTabs",
        "MuiMobileStepper",
        "MuiCircularProgress",
        "MuiGridList",
        "MuiHeroHeader"
      ],
      "pos": 12,
      "required": true
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiGrid",
  "component_group_uuid": null
}