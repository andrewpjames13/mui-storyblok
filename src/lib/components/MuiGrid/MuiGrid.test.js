import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiGrid from './MuiGrid';

function setup() {
  const props = {
    content: [{
      component: 'MuiGridItem',
      content: [{
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],
  };
  const comp = shallow(<MuiGrid {...props} />);
  return { comp, props };
}

describe('<MuiGrid />', () => {
  it('renders MuiGrid', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiGrid {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
