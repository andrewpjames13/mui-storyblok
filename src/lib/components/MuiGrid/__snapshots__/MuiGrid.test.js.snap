// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiGrid /> snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-3 MuiIcon-colorSecondary"
    >
      android
    </span>
  </div>
</div>
`;
